buildkitd/dockerd-wrapper.sh

```diff
@@ -193,8 +193,18 @@ load_registry_images() {
     if [ -n "$EARTHLY_DOCKER_LOAD_REGISTRY" ]; then
         echo "Loading images from BuildKit via embedded registry..."
         for img in $EARTHLY_DOCKER_LOAD_REGISTRY; do
-            user_tag=$(printf '%s' "$img" | cut -d'/' -f2-)
-            with_reg="$buildkit_docker_registry/$img"
+            case "$img" in
+                *'|'*)
+                    with_reg="$buildkit_docker_registry"/$(printf '%s' "$img" | cut -d'|' -f1)
+                    user_tag=$(printf '%s' "$img" | cut -d'|' -f2-)
+                    ;;
+                *)
+                    # Old format before v0.6.21.
+                    with_reg="$buildkit_docker_registry/$img"
+                    user_tag=$(printf '%s' "$img" | cut -d'/' -f2-)
```

Comment:
```suggestion
                    user_tag="$(printf '%s' "$img" | cut -d'/' -f2-)"
```

Reply:
I think it doesn't work due to the `"$img"` in the middle - not what's the correct way to quote this (or if it even needs quotes in this case)

Reply:
that surprises me, we have similar bits of shell that use double quotes both around and inside a shell out: https://github.com/earthly/earthly/blob/main/tests/secret-provider-config/test-secret-provider#L25

Reply:
anyways, if it's passing our linter and tests work might be good enough.

Reply:
TIL!

Fixed